docs/internal/compiler-steps.md

```diff
@@ -0,0 +1,119 @@
+# Compiler Steps
+
+## Overview
+
+This is a preliminary study on each compiler step to produce executable objects from Verona code.
+The initial objective of the Verona compiler is to generate a single executable from a number of files, directories and existing foreing objects.
+Loadable code and separate compilation processes are not taken into account in this document but are under discussion and may be introduced in the future.
+
+### Verona Sources
+
+The Verona compiler treats all files inside the same directory as a single module.
+Point the compiler to the directory and it will compile all (Verona) files together and include other modules (via `use` or `type`) in the same way.
+
+The end result is a single meta-module containing all modules: the main directory and all included modules from any module.
+This final module must contain a `Main` class with a `main(env : Env) : I32` method on any of the sub-modules, that will be the entry point for the program.
+Having zero or more than one `Main` classes is a compiler error.
```

Comment:
Not vitally important but it's nice if you can provide a compiler flag to specify a non-default main class, so that you can write stand-alone tests in a module and build an executable for each test by just telling the compiler that each test entry point is the one to use.

Reply:
I think this is orthogonal. Whatever `Main` is, having zero or more than one is still a compiler error. That was the point I was trying to get across here.

Reply:
I've added a phrase to mention that.